drivers/lte_link_control/lte_lc.c

```diff
@@ -166,24 +174,55 @@ static int w_lte_lc_init(void)
 
 static int w_lte_lc_connect(void)
 {
-	if (at_cmd_write(network_mode, NULL, 0, NULL) != 0) {
-		return -EIO;
-	}
+	int err;
+	char *current_network_mode = (char *)nw_mode_preferred;
```

Comment:
Here you need a cast because `nw_mode_preferred` is `const char []`, so why not make `current_network_mode ` a `const char *` to get rid of the cast, instead of casting down `nw_mode_preferred` discarding its const modifier? Sometimes it's dangerous to do that 💣

Reply:
You will also no longer need the cast on line 206.

Reply:
Agreed, avoiding the danger and updated per your suggestion